container.go

```diff
@@ -838,6 +838,32 @@ func (c *Client) RemoveContainer(opts RemoveContainerOptions) error {
 	return nil
 }
 
+// PutContainerArchiveOptions is the set of options that can be used when
+// uploading an archive into a container.
+//
+// See https://goo.gl/Ss97HW for more details.
+type PutContainerArchiveOptions struct {
+	InputStream          io.Reader `json:"-" qs:"-"`
+	Container            string    `json:"-" qs:"-"`
+	Path                 string    `qs:"path"`
+	NoOverwriteDirNonDir bool      `qs:"noOverwriteDirNonDir"`
+}
+
+// PutContainerArchive uploads a tar archive to be extracted to a path in the
+// filesystem of the container.
+//
+func (c *Client) PutContainerArchive(opts PutContainerArchiveOptions) error {
+	if opts.Container == "" {
+		return &NoSuchContainer{ID: opts.Container}
+	}
+	url := fmt.Sprintf("/containers/%s/archive", opts.Container)
```

Comment:
just combine this into one line, or just do it inline in the c.stream(...) call.